src/Features/Core/Portable/CodeFixesAndRefactorings/CodeActionRequestPriorityProvider.cs

```diff
@@ -0,0 +1,129 @@
+        {
+            var priority = provider.Priority;
+
+            // If caller isn't asking for prioritized result, then run all analyzers.
+            if (priority == CodeActionRequestPriority.None)
+                return true;
+
+            // 'CodeActionRequestPriority.Lowest' is used for suppression/configuration fixes,
+            // which requires all analyzer diagnostics.
+            if (priority == CodeActionRequestPriority.Lowest)
+                return true;
+
+            // The compiler analyzer always counts for any priority.  It's diagnostics may be fixed
+            // by high pri or normal pri fixers.
+            if (analyzer.IsCompilerAnalyzer())
+                return true;
+
+            // Check if we are computing diagnostics for 'CodeActionRequestPriority.Low' and
+            // this analyzer was de-prioritized to low priority bucket.
+            if (priority == CodeActionRequestPriority.Low &&
+                provider.IsDeprioritizedAnalyzerWithLowPriority(analyzer))
+            {
+                return true;
+            }
+
+            // Now compute this analyzer's priority and compare it with the provider's request 'Priority'.
+            // Our internal 'IBuiltInAnalyzer' can specify custom request priority, while all
+            // the third-party analyzers are assigned 'Normal' priority.
+            var analyzerPriority = analyzer is IBuiltInAnalyzer { RequestPriority: var requestPriority }
+                ? requestPriority
+                : CodeActionRequestPriority.Normal;
+
+            return priority == analyzerPriority;
+        }
+
+        /// <summary>
+        /// Returns true if the given <paramref name="codeFixProvider"/> should be considered a candidate when computing
+        /// fixes for the given <see cref="ICodeActionRequestPriorityProvider.Priority"/>.
+        /// </summary>
+        public static bool MatchesPriority(this ICodeActionRequestPriorityProvider provider, CodeFixProvider codeFixProvider)
+        {
+            if (provider.Priority == CodeActionRequestPriority.None)
+            {
+                // We are computing fixes for all priorities
+                return true;
+            }
+            if (provider.Priority == CodeActionRequestPriority.Low)
+            {
+                // 'Low' priority can be used for two types of code fixers:
+                //  1. Those which explicitly set their 'RequestPriority' to 'Low' and
+                //  2. Those which can fix diagnostics for expensive analyzers which were de-prioritized
+                //     to 'Low' priority bucket to improve lightbulb population performance.
+                // Hence, when processing the 'Low' Priority bucket, we accept fixers with any RequestPriority,
+                // as long as they can fix a diagnostic from an analyzer that was executed in the 'Low' bucket.
+                return true;
+            }
```

Comment:
was this codepath tested?  i feel like i saw good testing around analyzer prioritization, but it wasn't clear if code fix prioritization was tested.  if so, can you link me to the part of the test that does this?